config/servicemappings/edgecontainer.yaml

```diff
@@ -0,0 +1,136 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: core.cnrm.cloud.google.com/v1alpha1
+kind: ServiceMapping
+metadata:
+  name: edgecontainer.cnrm.cloud.google.com
+  namespace: cnrm-system
+spec:
+  name: Edgecontainer
+  version: v1beta1
+  serviceHostName: edgecontainer.googleapis.com
+  resources:
+    - name: google_edgecontainer_cluster
+      kind: EdgecontainerCluster
+      idTemplate: "projects/{{project}}/locations/{{location}}/clusters/{{name}}"
+      idTemplateCanBeUsedToMatchResourceName: false
+      resourceAvailableInAssetInventory: false
+      metadataMapping:
+        name: name
+        labels: labels
+      resourceID:
+        targetField: name
+      hierarchicalReferences:
+        - type: project
+          key: projectRef
+      resourceReferences:
+        - tfField: authorization.admin_users.username
+          key: serviceAccountRef
```

Comment:
I am thinking since the service account reference will be created and managed trough KCC, it seems more easy for user to understand as service account?

Reply:
I think we want to follow the guide to maintain consistency. This name represents a reference to 'username' field, so I think usernameRef makes more sense.

Reply:
Make sense, thanks.